core/src/main/java/com/datastax/oss/driver/api/core/cql/BatchStatement.java

```diff
@@ -131,6 +219,15 @@ static BatchStatement newInstance(
         Statement.NO_NOW_IN_SECONDS);
   }
 
+  /**
+   * Returns a builder to create an instance of the default implementation.
+   *
+   * <p>Note that this builder is mutable and not thread-safe.
+   */
+  static BatchStatementBuilder builder() {
+    return new BatchStatementBuilder(BatchType.UNKNOW);
+  }
```

Comment:
I don't think we want to do this at all.  I mentioned this in comments below but the current builder interface on this class already requires users to provide a non-null BatchType in order to get a builder.  That seems like exactly the kind of thing we want to encourage.  We should only worry about fixing the newInstance() methods above, and I provided some feedback on a way to do that that doesn't require the introduction of an UNKNOWN batch type.

Reply:
Thanks , I aggree that introduce an UNKNOW type is not a good idea.